popup.js

```diff
@@ -135,45 +141,53 @@ function minimizeStyles(message) {
  * @param {Document} doc The Document that contains the rendered HTML.
  * @param {Element} element The Element whose style attributes should be
  *     minimized.
+ * @param {string} id The id of the Element in the final page.
  * @param {number} index The index in |message.html| where the Element's style
  *     attribute is specified.
  */
-function minimizeStyle(message, doc, element, index) {
-  var originalStyle = element.getAttribute('style');
-  element.removeAttribute('style');
-  var unstyledStyle = doc.defaultView.getComputedStyle(element, null);
-
-  var fullStyleDeclaration = message.html[index];
-  var stylePrefix;
-  var styleSuffix;
-  // Remove style declaration from style attribute.
-  var style = fullStyleDeclaration.replace(
-    /^style=("|&(amp;)*?quot;)(.*)("|&(amp;)*?quot;) ?$/,
-    function(match, quote, p2, content) {
-      stylePrefix = 'style=' + quote;
-      styleSuffix = quote + ' ';
-      return content;
+function minimizeStyle(message, doc, element, id, index) {
+  var originalStyleAttribute = element.getAttribute('style');
+  var originalStyleMap = message.idToStyleMap[id];
+  var requiredStyleMap = {};
+  var maxNumberOfIterations = 5;
+
+  for (var i = 0; i < maxNumberOfIterations; i++) {
```

Comment:
This is a fairly complex algorithm, can you document it above this line in 3-4 sentences?

Something like... We compare the computed style before and after removing the style attribute and accumulate the differences in |requiredStyleMap|. Because some properties affect other properties, such as border-style: solid causing a change in border-width, we do this iteratively until a fixed-point is reached (or |maxNumberOfIterations| is hit).